pgjdbc/src/main/java/org/postgresql/jdbc/PgArray.java

```diff
@@ -175,6 +176,15 @@ public Object getArrayImpl(long index, int count, Map<String, Class<?>> map) thr
       return null;
     }
 
+    if (oid == Oid.JSON) {
+      try {
+        return jsonToArray(fieldString);
```

Comment:
you can inline the method `jsonToArray()`, because the `null` check is already done in line 176 above